src/pldm_fw.cpp

```diff
@@ -113,77 +215,71 @@ types::PldmPacket
 void PldmFramework::sendPanelFunctionToPhyp(
     const types::FunctionNumber& funcNumber)
 {
-    types::PdrList pdrs =
-        utils::getPDR(phypTerminusID, frontPanelBoardEntityId,
-                      stateIdToEnablePanelFunc, "FindStateEffecterPDR");
-
-    if (pdrs.empty())
+    pldm_instance_id_t instance = 0;
+    try
     {
-        std::map<std::string, std::string> additionalData{};
-        additionalData.emplace("DESCRIPTION",
-                               "Empty PDR returned for panel entity id.");
-        utils::createPEL("com.ibm.Panel.Error.HostCommunicationError",
-                         "xyz.openbmc_project.Logging.Entry.Level.Warning",
-                         additionalData);
-        return;
-    }
+        // Get PDR for Panel
+        types::PdrList pdrs =
+            utils::getPDR(phypTerminusID, frontPanelBoardEntityId,
+                          stateIdToEnablePanelFunc, "FindStateEffecterPDR");
 
-    types::Byte instance = getInstanceID();
+        if (pdrs.empty())
+        {
+            throw std::runtime_error(
+                "Empty PDR returned for front panel board entity.");
+        }
 
-    types::PldmPacket packet =
-        prepareSetEffecterReq(pdrs, instance, funcNumber);
+        // Allocate instance ID
+        instance = getInstanceID();
 
-    if (packet.empty())
-    {
-        std::map<std::string, std::string> additionalData{};
-        additionalData.emplace(
-            "DESCRIPTION", "pldm:SetStateEffecterStates request message empty");
-        utils::createPEL("com.ibm.Panel.Error.HostCommunicationError",
-                         "xyz.openbmc_project.Logging.Entry.Level.Warning",
-                         additionalData);
-        return;
-    }
+        // Prepare the message packet
+        types::PldmPacket packet =
+            prepareSetEffecterReq(pdrs, instance, funcNumber);
 
-    int fd = pldm_open();
-    if (fd == -1)
+        if (packet.empty())
+        {
+            throw std::runtime_error(
+                "pldm:SetStateEffecterStates request message empty");
+        }
+
+        // Open PLDM socket based communication
+        [[maybe_unused]] int fd = openPldm();
```

Comment:
Am I missing something? I don't see "fd" being used anywhere. Why are we declaring it?

Reply:
we need it for error logging.. will update